File: test/benchmarks/python/primes.py
from time import process_time as clock

def is_prime(n):
    if n < 2 : return False
    for i in range(2, n):
        if n % i == 0 : return False
    return True

def main():
    start = clock()

    N = 30000
    primes = []

    for i in range(N):
        if is_prime(i):
            primes.append(i)

    seconds = clock() - start
    print("elapsed: " + str(seconds) + "s | " + str(primes[-1]))

main()

File: test/benchmarks/node/primes.js
const process = require('process')

function isPrime(n) {
  if (n < 2) return false
  const minus = n - 1
  for (let i = 2; i < minus; i++) {
    if (n % i === 0) return false
  }
  return true
}

const start = process.hrtime.bigint()

const N = 30000
const primes = []

for (let i = 0; i < N; i++) {
  if (isPrime(i)) {
    primes.push(i)
  }
}

const seconds = Number(process.hrtime.bigint() - start) / 1000000000
console.log('elapsed: ' + seconds + 's | ' + primes[primes.length - 1])
